spec/services/save_and_invite_provider_user_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe SaveAndInviteProviderUser do
+  let(:provider) { create(:provider) }
+  let(:form) {
+    SupportInterface::ProviderUserForm.new(
+      email_address: 'test+invite_provider_user@example.com',
+      first_name: 'Firstname',
+      last_name: 'Lastname',
+      provider_ids: [provider.id],
+    )
+  }
+  let(:permissions) { { manage_users: [provider.id] } }
+  let(:provider_user) { form.build }
+  let(:save_service) { SaveProviderUser.new(provider_user: provider_user, permissions: permissions) }
+  let(:invite_service) { InviteProviderUser.new(provider_user: provider_user) }
```

Comment:
If we're making these injectable, might we as well inject stubs implementing `save!`? Slightly less coupling

Reply:
:+1: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1866/commits/2819a377045a5bb37eb6d96ca0270ede0e0690dc